typed-store/src/rocks/mod.rs

```diff
@@ -300,6 +300,35 @@ where
 
         Values::new(db_iter)
     }
+
+    /// Returns a vector of values corresponding to the keys provided.
+    fn multi_get(&self, keys: &[K]) -> Result<Vec<Option<V>>> {
+        let config = bincode::DefaultOptions::new()
+            .with_big_endian()
+            .with_fixint_encoding();
+
+        let mut keys_bytes = Vec::with_capacity(keys.len());
+        for k in keys.iter() {
+            let serialized_key = config.serialize(k)?;
+            // fallible call makes use of iterator hard
+            keys_bytes.push((self.cf(), serialized_key));
+        }
+
+        let results = self
+            .rocksdb
+            .multi_get_cf(keys_bytes.iter().map(|(cf, k)| (cf, k)));
+
+        let mut values_parsed = Vec::with_capacity(keys.len());
+        for value_byte in results {
+            // fallible call makes use of iterator hard
+            values_parsed.push(match value_byte? {
+                Some(data) => Some(bincode::deserialize(&data)?),
+                None => None,
+            });
+        }
```

Comment:
You're looking for the "collecting into a result" pattern: https://doc.rust-lang.org/rust-by-example/error/iter_result.html#fail-the-entire-operation-with-collect
```suggestion
        results.flat_map(|opt_val| opt_val.map(|data| bincode::deserialize(&data)).transpose()).collect()
```

Reply:
Have a look at how I changed it to use iterators. Its something in the middle, where I use ? within the iterator to make things less magic.

Reply:
Very nice job!